### PR TITLE
 feat(dune): support `+` prefixes for PPX CLI flags 

### PR DESCRIPTION
**Pull request by anmonteiro**
this is useful for cases like `melange.ppx` which emit alerts and allow to configure them via `-alert ...`.

### CODE REVIEW

**Comment by anmonteiro** [comment]
Why do you think this needs to be versioned? 

I believe the previous behavior would just plainly return an error, so we shouldn't have anyone depending on these features, right?

**Comment by rgrinberg** [comment]
Suppose that somebody would try to use this feature in a publicly released project. That person would need to remember to update the constraints in their project file to require at least dune 3.17. Without the version check, it is very likely that the person would forget to add this constraint, or they might even do it incorrectly, if they misremember which version of dune introduced this feature. Downstream user of this project might then encounter build failures should they build with an older version of dune. If we add a version check, we make this scenario more or less impossible.

It's true that the above argument is applicable to basically any user visible change in dune, but I think it works best for features. As for bug fixes, it's good that users can benefit from those without opting in. I think that this PR is certainly more of a feature than a bug fix.

**Comment by anmonteiro** [comment]
absolutely, thanks for the explainer!

**Comment by anmonteiro** [comment]
@rgrinberg ready for another look.

File: src/dune_rules/preprocess.ml [hunk 1]
```diff
@@ -12,9 +12,13 @@ module Pps_and_flags = struct
...
-        match String_with_vars.is_prefix ~prefix:"-" s with
-        | Yes -> Right s
-        | No | Unknown _ ->
+        match
+          ( String_with_vars.is_prefix ~prefix:"-" s
+          , String_with_vars.is_prefix ~prefix:"+" s )
+        with
+        | Yes, _ -> Right s
+        | _, Yes when syntax_version >= (3, 18) -> Right s
+        | (No | Unknown _), _ ->
```
Comment (rgrinberg) [inline src/dune_rules/preprocess.ml:21]: I think you could improve the error message here with a hint for `prefix = "+"`  when `syntax_version < (3, 18)`

**Review by rgrinberg — APPROVED** [review]

**Pull request merged**